### ISSUE TITLE
Dependency as dataclass attribute fails on 3.11+

### ISSUE THREAD

**Issue by eykd** [ZechCodes/Bevy#7]
This code fails in 3.11+:

```python
@dataclass
class Bar:
    value = 'bar'

@dataclass
class Foo:
    bar: Bar = dependency()
```

Here's a gist with a tox+pytest setup:

https://gist.github.com/eykd/0ddbc8ea7080c1702d95838c0d6e1f02

Test passes in 3.10, but fails 3.11 and 3.12.

Tail end of Tox output:

```
E           ValueError: mutable default <class 'test_bevy.test_it_should_instantiate_foo.<locals>.Bar'> for field bar is not allowed: use default_factory

/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/dataclasses.py:852: ValueError
============================================================================= short test summary info ==============================================================================
FAILED test_bevy.py::test_it_should_instantiate_foo - ValueError: mutable default <class 'test_bevy.test_it_should_instantiate_foo.<locals>.Bar'> for field bar is not allowed: use default_factory
================================================================================ 1 failed in 0.04s =================================================================================
py312: exit 1 (0.20 seconds) ~/tmp/bevy> pytest pid=2302
  py310: OK (0.22=setup[0.06]+cmd[0.15] seconds)
  py311: FAIL code 1 (0.22=setup[0.00]+cmd[0.21] seconds)
  py312: FAIL code 1 (0.20=setup[0.01]+cmd[0.20] seconds)
  evaluation failed :( (0.70 seconds)
```

**Comment by ZechCodes** [comment]
I don't know if my solution for #9 will resolve this. I'll have to dig into it later.

**Comment by ZechCodes** [comment]
The solution for #9 would resolve this issue. It's the premature injection that's causing the value error.

**Comment by ZechCodes** [comment]
be9566395b1466bd36828bda4b07ce1ffa8d57b2 resolves this issue and #9
